lib/OpenQA/WebAPI/Controller/API/V1/Table.pm

```diff
@@ -397,9 +397,16 @@ sub _prepare_settings {
     my @keys;
     if ($hp->{settings}) {
         for my $k (keys %{$hp->{settings}}) {
-            $k = trim $k;
             my $value = trim $hp->{settings}->{$k};
-            $k =~ s/[^\]\[0-9a-zA-Z_\+]//g;
+            $k = trim $k;
+            my $invalid_chars_in_key = $k;
+            # Remove any allowed chars so that only invalid chars remain
+            $invalid_chars_in_key =~ s/[\]\[0-9a-zA-Z_\+]//g;
+            $invalid_chars_in_key =~ s/(.)(?=.*\1)//g;    # Remove any chars that occurs more than once
+            if ($invalid_chars_in_key ne '') {
+                my $eick = join(', ', map { '<b>' . xml_escape($_) . '</b>' } split('', $invalid_chars_in_key));
+                return sprintf('Invalid characters %s in settings key <b>%s</b>', $eick, xml_escape($k));
+            }
```

Comment:
```suggestion
            my %invalid;
            $k =~ s/([^\]\[0-9a-zA-Z_\+])/$inv{$1} = 1/eg;
            delete $invalid{' '}; # silently remove spaces
            if (keys %invalid) {
                my $eick = join ', ', map { '<b>' . xml_escape($_) . '</b>' } sort keys %invalid;
                return sprintf('Invalid characters %s in settings key <b>%s</b>', $eick, xml_escape($k));
            }
```

Reply:
Also, if you want to keep your code, as an alternative to a regex:
```
my $invalid_chars_in_key = $k = tr/[]0-9a-zA-Z_+//dr;
```

Reply:
hm - this doesn't work as expected as it modifies `$k`.

Reply:
also I'm not sure if we want to `delete $invalid{' '}; # silently remove spaces`

Reply:
I updated it accordingly.